atox/src/main/kotlin/ui/user_profile/UserProfileFragment.kt

```diff
@@ -155,23 +168,82 @@ class UserProfileFragment : BaseFragment<FragmentUserProfileBinding>(FragmentUse
     }
 
     private fun createQrCodeDialog(): AlertDialog {
+        val qrCode = ImageView(requireContext())
+        qrCode.setPadding(qrCodePadding.asPx(resources).px.toInt())
+        createQrCode(imageView = qrCode)
+
+        return AlertDialog.Builder(requireContext())
+            .setTitle(R.string.tox_id)
+            .setView(qrCode)
+            .setPositiveButton(getString(R.string.share)) { _, _ ->
+                vm.viewModelScope.launch {
+                    val qrImageUri = getQrForSharing("tox_id_qr_code")
+                    val shareIntent = Intent().apply {
+                        action = Intent.ACTION_SEND
+                        clipData = ClipData.newRawUri(null, qrImageUri)
+                        type = "image/png"
+                        putExtra(Intent.EXTRA_STREAM, qrImageUri)
+                        flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
+                    }
+                    startActivity(Intent.createChooser(shareIntent, getString(R.string.tox_id_share)))
+                }
+            }
+            .setNegativeButton(getString(R.string.close)) { dialog, _ ->
+                dialog.dismiss()
+            }
+            .create()
+    }
+
+    /*
+     * Function will create a QR code for the Tox ID and assign it to the image view if specified.
```

Comment:
No need for this comment if we just return the QR code as suggested in another comment.